data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java

```diff
@@ -125,6 +166,42 @@ public void testEqualityDeletes() throws IOException {
     Assert.assertEquals("Table should contain expected rows", expected, actual);
   }
 
+  @Test
+  public void testEqualityDateDeletes() throws IOException {
+    initTable2();
+
+    Schema deleteRowSchema = table2.schema().select("*");
+    Record dataDelete = GenericRecord.create(deleteRowSchema);
+    List<Record> dataDeletes = Lists.newArrayList(
+            dataDelete.copy("dt", LocalDate.parse("2021-09-01"), "data", "a", "id", 1),
+            dataDelete.copy("dt", LocalDate.parse("2021-09-02"), "data", "b", "id", 2),
+            dataDelete.copy("dt", LocalDate.parse("2021-09-03"), "data", "c", "id", 3)
+    );
+
+    DeleteFile eqDeletes = FileHelpers.writeDeleteFile(
+            table2, Files.localOutput(temp.newFile()), Row.of(0), dataDeletes, deleteRowSchema);
+
+    table2.newRowDelta()
+            .addDeletes(eqDeletes)
+            .commit();
+
+    StructLikeSet expected = rowSetWithoutIds2(1, 2, 3);
+    StructLikeSet expectedSet = StructLikeSet.create(table2.schema().asStruct());
+
+    Iterables.addAll(expectedSet, expected.stream()
+            .map(record -> new InternalRecordWrapper(table2.schema().asStruct()).wrap(record))
+            .collect(Collectors.toList()));
+
+    StructLikeSet actual = rowSet(tableName2, table2, "*");
+    StructLikeSet actualSet = StructLikeSet.create(table2.schema().asStruct());
+
+    Iterables.addAll(actualSet, actual.stream()
+            .map(record -> new InternalRecordWrapper(table2.schema().asStruct()).wrap(record))
+            .collect(Collectors.toList()));
```

Comment:
Why are you using `InternalRecordWrapper` here and for the expected set? I think that `StructLikeSet` should work without it, right? If not, then we need to update the set that is produced by `rowSet` to do the wrapping since future tests may not know to wrap.

Reply:
@xloya, can you briefly describe what you did to address this?

Reply:
@rdblue OK. The data returned by the original `rowSetWithoutIds()`method of `DeleteReadTests` and the original `rowSet()` method of `TestGenericReaderDeletes` is the `GenericRecord` class. Calling the `Assert.assertEquals()` method will compare the two `StructLikeSets`. If it is `DATE/TIMESTAMP` type of the data, it will throw an exception of `Not a instance of Integer/Long`, so it needs to be converted to the `InternalRecordWrapper` class for comparison

Reply:
Maybe i should encapsulate the assertion method?

Reply:
I think either the conversion at the `assertion` or the conversion at `rowSetWithoutIds()` and `rowSet()` is fine, but it must be converted.